docker/cache/populate_cache.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+set -xuo pipefail
+
+if (( $# != 2 )); then
+  echo >&2 "usage: $0 [pex dev cache dir] [tox env][,tox env]*"
+  echo >&2 "Expected 2 arguments, got $#: $*"
+  exit 1
+fi
+
+function run_tox() {
+  local env="$1"
+  tox -e "${env}" -- --color --devpi --require-devpi -vvs
+  if (( $? == 42 )); then
+    echo >&2 "tox -e ${env} failed to start or connect to the devpi-server, exiting..."
+    exit 1
+  elif (( $? != 0 )); then
+    echo >&2 "tox -e ${env} failed, continuing..."
```

Comment:
This is very deliberate. If 2 tests fail out of ~500 (common enough when running without a devpi-server cache!), the devpi-server cache is likely ~fully populated for the run and we got what we needed. Compound this with ~19 permutations of IT being run through this - a ~4 hour process on a fast machine - and `set -e` would lead to no cache image ever successfully building.

Reply:
So the idea is to manually eyeball that the failures were only a small % when building the data-only image?

Reply:
Not really (that's hard to do if you have a docker that uses BuildKit anyway with its default fancy tui logging). I just compare sizes of the dev cache:
```
$ ./dtox.sh inspect
jsirois@db935e314f12:/development/pex$ du -shc /development/pex_dev/*
844M    /development/pex_dev/devpi
1.6G    /development/pex_dev/pyenv
2.4G    total
```
844M is a little bigger than my local dev cache, which makes sense since I don't always run all tox environments locally and which tests are run or what exactly is downloaded does vary a little bit by interpreter version.

When automating the cache image build, I'm guessing no action will be needed in practice, the failure rates seen in the wild in ITs are O(1/500) when an IT network flakes and not all IT runs in CI do (most don't). If I do take action, I think a cache size comparison to the prior cache image - maybe no more than 1% shrinkage, would be the easiest check that is both somewhat robust and tunable in an easy way.